### ISSUE TITLE
XML generated for absent date is rejected by xml2rfc --v3

### ISSUE THREAD

**Issue by jyasskin** [cabo/kramdown-rfc2629#64]
I have the following informative reference in my markdown file:

```yaml
informative:
  ServiceWorkers:
    target: https://w3c.github.io/ServiceWorker/
    title: "Service Workers Nightly"
    seriesinfo:
      W3C: "ED"
```

It's missing a `date` entry, so kramdown-rfc2629 generates `<date year="n.d."/>` in the XML. When I configure [i-d-template](https://github.com/martinthomson/i-d-template/) with `XML2RFCV3=true` to use `xml2rfc --v3`, I get the error:

```
Error: Expected <date> attribute "year" to be an integer, but found "n.d."
```

https://tools.ietf.org/html/rfc7991#section-2.17 and https://tools.ietf.org/html/draft-iab-xml2rfc-v3-bis-00#section-2.17 say prose text is expected to be supported, so maybe kramdown-rfc2629 is doing the right thing, but https://tools.ietf.org/html/draft-levkowetz-xml2rfc-v3-implementation-notes-10#section-4.1.3.2 claims prose text in the `year` attribute is unworkable and suggests it be in the content instead.

Maybe kramdown-rfc2629 should emit its own error for an absent year.

**Comment by cabo** [comment]
In kramdown-rfc + xml2rfc, you can suppress the date attributes entirely by setting it to false:

  date: false

That works in v2, but with v3 you will run into a new bug of xml2rfc in v3 mode where it puts in the current date instead.  But that is being fixed as we speak: https://trac.tools.ietf.org/tools/xml2rfc/trac/ticket/450

(And I need to push my more v3-friendly branch of kramdown-rfc2629.)

I’m not sure foisting the n.d. convention on the kramdown-rfc users was very bright; the RFC editor doesn’t like it, but other citation styles require it for undated material, e.g., https://apastyle.apa.org/learn/faqs/cite-website-material
https://apastyle.apa.org/style-grammar-guidelines/references/missing-information

Grüße, Carsten

**Comment by jyasskin** [comment]
Adding an explicit `date: false` to the reference also produces the `<date year="n.d."/>` for me, with version 1.2.12.

**Comment by cabo** [comment]
Can’t reproduce (and this has been a feature for a long time).
Can you supply your input?

Grüße, Carsten

**Comment by cabo** [comment]
Good idea.  Done.

Leaving this issue open as we still need a workaround if text dates are no longer to be accepted in the `year` attribute in v3.

**Comment by cabo** [comment]
I neglected to report here that issue 450 is fixed in XML2RFC release 2.24.0 on August 10, 2019 (?), and the compatibility issue underlying this bug report was fixed in https://trac.tools.ietf.org/tools/xml2rfc/trac/changeset/3573 on 2020-01-21, included in XML2RFC 2.39.0 on 2020-01-31.

**Comment by Himanshu4521** [comment]
True
